packages/teleterm/src/ui/commandLauncher.ts

```diff
@@ -92,6 +92,50 @@ const commands = {
     },
   },
 
+  'tsh-install': {
+    displayName: '',
+    description: '',
+    run(ctx: IAppContext) {
+      ctx.mainProcessClient.symlinkTsh().then(
+        isSymlinked => {
+          if (isSymlinked) {
+            ctx.notificationsService.notifyInfo(
+              'tsh successfully installed in PATH'
+            );
+          }
+        },
+        error => {
+          ctx.notificationsService.notifyError({
+            title: 'Could not install tsh in PATH',
+            description: `Ran into an error: ${error}`,
```

Comment:
I didn't check it, but it seems to me that the `error` here is not a plain string, but an object with a `message` field (as we throw `error` from the IPC handler), so we will get a runtime error.

Reply:
> I didn't check it, but it seems to me that the `error` here is not a plain string, but an object with a `message` field (as we throw `error` from the IPC handler), so we will get a runtime error.

Errors that pass through IPC or the context bridge are still instances of `Error` with `message`, they're just stripped of any custom properties that might be attached to them. See [the _A word on error handling_ section](https://www.electronjs.org/docs/latest/tutorial/ipc#1-listen-for-events-with-ipcmainhandle).

The place where we discussed returning plain objects instead of errors was tshd client but we never got around to that (gravitational/webapps.e#309) 😏.

And I don't think we'd get a runtime error in any case:

```typescript
const err = {message: "test"}; `err: ${err}`
// > "err: [object Object]"
```

Reply:
>Errors that pass through IPC or the context bridge are still instances of Error with message, they're just stripped of any custom properties

I see, I was sure that name is passed too, as it is standard `Error` property. 

>And I don't think we'd get a runtime error in any case

Ah right, `Objects are not valid as a react child` happens only when rendering objects directly in JSX.